### PR TITLE
Update Logging

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR does two things:
- Update the calls to the `logging` library to use printf-style formatting.
- Use log group in the Actions log.

## 💭 Motivation and context ##

We prefer printf-style formatted log messages for our Python code due to the internals of the library.

Grouping the log output for each repo makes it easier to browse the logs when trying to troubleshoot.

## 🧪 Testing ##

Automated tests pass. You can see the result in the logs in [this Action run](https://github.com/cisagov/action-lineage/runs/3898060846?check_suite_focus=true).

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

File: src/lineage/entrypoint.py [hunk 1]
```diff
@@ -352,44 +358,49 @@ def main() -> None:
...
         for lineage_id, v in config["lineage"].items():
-            logging.info(f"Processing lineage: {lineage_id}")
+            logging.info("Processing lineage: %s", lineage_id)
             try:
                 local_branch = v.get("local-branch")
                 remote_branch = v.get("remote-branch", "HEAD")
                 remote_url = v["remote-url"]
             except KeyError as e:
-                logging.warning(f"Missing config key while reading {lineage_id}: {e}")
+                logging.warning(
```
Comment (jsf9k) [inline src/lineage/entrypoint.py]: Should we use `logging.error` here?  That would include a stack trace.

Comment (mcdonnnj) [reply src/lineage/entrypoint.py]: Do you mean `logging.exception`?

Comment (jsf9k) [reply src/lineage/entrypoint.py]: Yes!

Comment (mcdonnnj) [reply src/lineage/entrypoint.py]: Please see ffb6f0b.

**Review by dav3r — APPROVED** [review]
Looking good! 🦞

**Review by jsf9k — APPROVED** [review]

**Pull request merged**